spark-cassandra-connector/src/test/scala/com/datastax/spark/connector/cql/TableDefSpec.scala

```diff
@@ -0,0 +1,72 @@
+package com.datastax.spark.connector.cql
+
+import com.datastax.spark.connector.types.{BigIntType, MapType, VarCharType, IntType}
+import org.scalatest.{WordSpec, Matchers}
+
+class TableDefSpec extends WordSpec with Matchers {
+
+  "A TableDef" should {
+    "produce valid CQL" when {
+      "it contains no clustering columns" in {
+        val column1 = ColumnDef("c1", PartitionKeyColumn, IntType)
+        val column2 = ColumnDef("c2", RegularColumn, VarCharType)
+        val tableDef = TableDef("keyspace", "table", Seq(column1), Seq.empty, Seq(column2))
+        tableDef.cql should be(
+          """CREATE TABLE "keyspace"."table" (
+            |  "c1" int,
+            |  "c2" varchar,
+            |  PRIMARY KEY (("c1"))
+            |)""".stripMargin
+        )
+      }
+
+      "it contains clustering columns" in {
+        val column1 = ColumnDef("c1", PartitionKeyColumn, IntType)
+        val column2 = ColumnDef("c2", RegularColumn, VarCharType)
+        val column3 = ColumnDef("c3", RegularColumn, VarCharType)
+        val tableDef = TableDef("keyspace", "table", Seq(column1), Seq(column2), Seq(column3))
+        tableDef.cql should be(
+          """CREATE TABLE "keyspace"."table" (
+            |  "c1" int,
+            |  "c2" varchar,
+            |  "c3" varchar,
+            |  PRIMARY KEY (("c1"), "c2")
+            |)""".stripMargin
+        )
+      }
+
+      "it contains compound partition key and multiple clustering columns" in {
+        val column1 = ColumnDef("c1", PartitionKeyColumn, IntType)
+        val column2 = ColumnDef("c2", RegularColumn, VarCharType)
```

Comment:
This test is confusing to me, shouldn't this throw an exception since c2 is marked as a regular column but then used in the Partition key?